src/uncrustify.cpp

```diff
@@ -1416,8 +1418,8 @@ static void add_func_header(c_token_t type, file_mem &fm)
 
       if (ref->type == CT_FUNC_DEF && ref->parent_type == CT_NONE && ref->next)
       {
-         int found_brace = 0; // Set if a close brace is found before a newline
-         while (ref->type != CT_NEWLINE && (ref = ref->next))
+         int found_brace = 0;                                   // Set if a close brace is found before a newline
+         while ((ref->type != CT_NEWLINE) && (ref = ref->next)) /* \todo is the assignment of ref wanted here?, better move it to the loop */
```

Comment:
This was done on purpose, it is one of those _tricks_ that allows to save space for decreased readability as a tradeoff. 
After the type check, `ref` is assigned to `ref->next` and then, because there is a assignment inside braces, the assigned to var (`ref`) is bool evaluated. As `ref` is a pointer it is checked for **not** being `NULL`.
It would be better to clear things up a bit.